lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,28 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # exercise car model
+  class Car
+    attr_reader :year, :color, :model, :current_speed
+    def self.default_car
+      Car.new {}
+    end
+
+    def initialize(args)
+      raise ArgumentError if args.class != Hash
+
+      @year ||= 2015
```

Comment:
in fact, you are not parsing data from args. Always assign default 2015.

@year = args.fetch(:year, 2015)